rules/naming/tests/ValueObjectFactory/PropertyRenameFactory/PropertyRenameFactoryTest.php

```diff
@@ -45,31 +42,36 @@ public function __construct(?string $name = null, array $data = [], $dataName =
     }
 
     /**
-     * @dataProvider createDataProvider
+     * @dataProvider provideData()
      */
-    public function testCreate(Property $property, string $expectedName, string $currentName): void
+    public function test(SmartFileInfo $fileInfoWithProperty, string $expectedName, string $currentName): void
     {
-        /** @var PropertyRename $actualPropertyRename */
+        $property = $this->getPropertyFromFileInfo($fileInfoWithProperty);
+
         $actualPropertyRename = $this->propertyRenameFactory->create($property);
+        $this->assertNotNull($actualPropertyRename);
 
-        $this->assertSame($property, $actualPropertyRename->getProperty());
+        /** @var PropertyRename $actualPropertyRename */
+        $this->assertSame($fileInfoWithProperty, $actualPropertyRename->getProperty());
         $this->assertSame($expectedName, $actualPropertyRename->getExpectedName());
         $this->assertSame($currentName, $actualPropertyRename->getCurrentName());
-        $this->assertInstanceOf(PropertyProperty::class, $actualPropertyRename->getPropertyProperty());
-        $this->assertInstanceOf(ClassLike::class, $actualPropertyRename->getClassLike());
     }
 
-    public function createDataProvider(): Iterator
+    public function provideData(): Iterator
     {
-        yield [$this->getPropertyFromFile(__DIR__ . '/FixtureBasic/SomeClass.php'), 'eliteManager', 'eventManager'];
+        yield [new SmartFileInfo(__DIR__ . '/Fixture/SomeClass.php.inc'), 'eliteManager', 'eventManager'];
```

Comment:
@dobryy Here is the root problem. Data provider should be just simple bare data.

The `getPropertyFromFile()` calls a sevices, that supposed to be set up in `setUp()` method. But that is `null`, right?

The problem is order of these method:

- first `dataProvider()` is called
- then `setUp()` - only now service actually exist
- then `test*()` method

A PHPStan rule proposed here https://github.com/symplify/symplify/issues/2226

Reply:
I see, I see, you're right before I created `getPropertyFromFile()` I used `setUp()`. I didn't want to put this logic into the test itself.

Reply:
Yea, this is crap in PHPUnit :/. I also expect `setUp()` being run before any other code.